eox_hooks/tasks.py

```diff
@@ -0,0 +1,66 @@
+"""
+Task file for eox-hooks plugin.
+"""
+import logging
+
+from celery import shared_task
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from opaque_keys.edx.keys import CourseKey
+
+from eox_hooks.edxapp_wrapper.enrollments import get_course_enrollment
+
+try:
+    from course_modes.models import CourseMode
+    modes_for_course = CourseMode.ALL_MODES
+except ImportError:
+    modes_for_course = ["honor", "professional", "verified", "audit", "no-id-professional", "credit", "masters"]
+
+
+ENROLLMENT_MAX_RETRIES = 3
+User = get_user_model()
+CourseEnrollment = get_course_enrollment()
+log = logging.getLogger(__name__)
+
+
+@shared_task(bind=True, max_retries=ENROLLMENT_MAX_RETRIES)
+def create_enrollments_for_program(self, username, followup_enrollments):
+    """
+    Create enrollments for a list of courses that define a Course Program.
+
+    Args:
+        username (str): identification of the user to be enrolled.
+        mode (str): CourseMode for the enrollment.
+        followup_enrollments ([Dict]): list of values used in the enrollment
+        creation. The valid attributes are:
+            - course_id
+            - mode
+    """
+    try:
+        user = User.objects.get(username=username)
+
+        for enrollment in followup_enrollments:
+
+            course_key_str = enrollment.pop("course_id", "")
+
+            if enrollment.get("mode", "audit").lower() not in modes_for_course:
+                log.error(
+                    "Couldn't create enrollment for %s because the configuration is not valid: invalid course mode.",
+                    course_key_str
+                )
+                continue
+
+            course_key = CourseKey.from_string(course_key_str)
+            enrollment["course_key"] = course_key
+
+            if course_key.org not in getattr(settings, "course_org_filter", []):
```

Comment:
@mariajgrimaldi  what do you think about checking the organization and course mode (46) in the same line and return a single error message with something like "invalid organization or course mode"?